src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -467,5 +475,77 @@ private static string GetTempAssemblyName(AssemblyName parent, string ns)
         {
             return parent.Name + ".XmlSerializers" + (ns == null || ns.Length == 0 ? "" : "." + ns.GetHashCode());
         }
+
+        private static void ParseReferences(string value, Dictionary<string, string> dictionary)
+        {
+            List<string> list = new List<string>();
+            if (value.Length > 0)
+            {
+                string[] entries = value.Split(new char[] { ';' });
+                for (int i = 0; i < entries.Length; i++)
+                {
+                    string entry = entries[i].Trim();
+                    if (entry == null || entry.Length == 0)
+                        continue;
+                    list.Add(entry);
+                }
+            }
+
+            foreach (var reference in list)
+            {
+                if (reference.EndsWith(".dll") || reference.EndsWith(".exe"))
+                {
+                    if (File.Exists(reference))
+                    {
+                        string filename = Path.GetFileNameWithoutExtension(reference);
+                        if (!string.IsNullOrEmpty(filename))
+                        {
+                            dictionary.Add(filename, reference);
+                        }
+                    }
+                }
+
+            }
+        }
+
+        private static Assembly SgenAssemblyResolver(object source, ResolveEventArgs e)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(references) || string.IsNullOrEmpty(e.Name) || e.Name.Split(',').Count() == 0)
+                {
+                    return null;
+                }
+
+                string assemblyname = e.Name.Split(',')[0];
+                if (string.IsNullOrEmpty(assemblyname))
+                {
+                    return null;
+                }
+
+                Dictionary<string, string> referencedic = new Dictionary<string, string>();
+                ParseReferences(references, referencedic);
+
+                string reference = referencedic[assemblyname];
+                if (!string.IsNullOrEmpty(reference))
```

Comment:
Dictionary this[] getter throws KeyNotFoundException if the assembly name isn't found in the dictionary. You should check .Contains(assemblyname) first and return null if it's not found. Also the dictionary is case sensitive. If a reference assembly name is passed on the command line with a different case than the reference is using you won't find it. Construct the dictionary passing StringComparer.OrdinalIgnoreCase to fix this.